cmk/base/plugins/agent_based/prism_containers.py

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+# Copyright (C) 2019 tribe29 GmbH - License: GNU General Public License v2
+# This file is part of Checkmk (https://checkmk.com). It is subject to the terms and
+# conditions defined in the file COPYING, which is part of this source code package.
+# ported by (c) Andreas Doehler <andreas.doehler@bechtle.com/andreas.doehler@gmail.com>
+from contextlib import suppress
+from typing import Any, Dict, Mapping
+
+from .agent_based_api.v1 import get_value_store, GetRateError, register, Service
+from .agent_based_api.v1.type_defs import CheckResult, DiscoveryResult, StringTable
+from .utils.df import df_check_filesystem_single, FILESYSTEM_DEFAULT_LEVELS
+from .utils.prism import load_json
+
+Section = Dict[str, Mapping[str, Any]]
+
+
+def parse_prism_container(string_table: StringTable) -> Section:
+    parsed: Section = {}
+    data = load_json(string_table)
+    for element in data.get("entities", {}):
+        parsed.setdefault(element.get("name", "unknown"), element)
+    return parsed
+
+
+register.agent_section(
+    name="prism_containers",
+    parse_function=parse_prism_container,
+)
+
+
+def discovery_prism_container(section: Section) -> DiscoveryResult:
+    for item in section:
+        yield Service(item=item)
+
+
+def check_prism_container(item: str, params: Mapping[str, Any], section: Section) -> CheckResult:
+    value_store = get_value_store()
+    data = section.get(item)
+    if not data:
+        return
+
+    capacity, freebytes = map(
+        int,
+        (
+            data["usageStats"].get("storage.user_capacity_bytes", 0),
+            data["usageStats"].get("storage.user_free_bytes", 0),
+        ),
+    )
+
+    with suppress(GetRateError):
+        yield from df_check_filesystem_single(
+            value_store,
+            item,
+            capacity / 1024**2,
+            freebytes / 1024**2,
+            0,
+            None,
+            None,
+            params=params,
+        )
+
+
+register.check_plugin(
+    name="prism_containers",
+    service_name="NTNX Container %s",
+    sections=["prism_containers"],
+    check_default_parameters=FILESYSTEM_DEFAULT_LEVELS,
+    discovery_function=discovery_prism_container,
+    check_function=check_prism_container,
+    check_ruleset_name="filesystem",
```

Comment:
This would be a breaking change. We remove the prism_container rule and replace it with the filesystem rule. Is there a sensible way to convert the two? I would like to see if we can make the migration from prism_container to filesystem part of our update-config script

Reply:
The usage of filesystem instead of the prism_container rule should be no problem. This was only converted as the original used it as it's own rule.

Reply:
If someone has configured the prism_container rule they need to manually change the rule after an update currently. I want to avoid that. Therefore the question how easy it would be to convert an old 'prism_container' rule into a 'filesystem' rule during the update.

Using filesystem here is fine with me.

Reply:
The old prism_container rule only has the option for warn and crit as a check_levels.
old --> params["levels"]
--
new --> params["levels"]
Booth times the levels are only a tuple for warn, crit. The only change is from group "prism_container" to "filesystem".

Reply:
we kept the old rule and set it to deprecated. This allows users to migrate after a major update.